src/apispec/ext/marshmallow/field_converter.py

```diff
@@ -328,23 +319,13 @@ def field2length(self, field, **kwargs):
         min_attr = "minItems" if is_array else "minLength"
         max_attr = "maxItems" if is_array else "maxLength"
 
-        for validator in validators:
-            if validator.min is not None:
-                if hasattr(attributes, min_attr):
-                    attributes[min_attr] = max(attributes[min_attr], validator.min)
-                else:
-                    attributes[min_attr] = validator.min
-            if validator.max is not None:
-                if hasattr(attributes, max_attr):
-                    attributes[max_attr] = min(attributes[max_attr], validator.max)
-                else:
-                    attributes[max_attr] = validator.max
-
-        for validator in validators:
-            if validator.equal is not None:
-                attributes[min_attr] = validator.equal
-                attributes[max_attr] = validator.equal
-        return attributes
+        equal_list = [
+            validator.equal for validator in validators if validator.equal is not None
+        ]
+        if equal_list:
+            return {min_attr: equal_list[0], max_attr: equal_list[0]}
+
```

Comment:
Both former and new code don't deal with the case of a user entering two incompatible equal constraints (or an equal constraint incompatible with a range). This is similar to #198 and I'm not sure we should do something about it.

The change introduced here is that if two different equal constraints are provided, we're keeping the last one while we used to keep the first. Just a note, I don't think it matters.